src/Runtime/Distributions/Automata/Automaton.Condensation.cs

```diff
@@ -77,7 +78,7 @@ public class Condensation
             /// <summary>
             /// The dictionary containing information associated with every state of the condensation.
             /// </summary>
-            private readonly Dictionary<int, CondensationStateInfo> stateIdToInfo;
+            private GenerationalDictionary<int, CondensationStateInfo> stateInfo;
```

Comment:
Why was "readonly" dropped?

Reply:
At some point GenerationalDictionary was a struct rather than class. But it brought more problems then benefits. Will return it.

Reply:
Fixed